api/client/proto/authservice.proto

```diff
@@ -1359,6 +1359,52 @@ message CreateRegisterChallengeRequest {
     DeviceType DeviceType = 2 [ (gogoproto.jsontag) = "device_type" ];
 }
 
+// ResourceType
+enum ResourceType {
+    // Default value implies the resource was not defined.
+    RESOURCE_TYPE_UNSPECIFIED = 0;
+    // RESOURCE_TYPE_DATABASE_SERVER the resource is DatabaseServer.
+    RESOURCE_TYPE_DATABASE_SERVER = 1;
+    // RESOURCE_TYPE_APPLICATION_SERVER the resource is AppServer.
+    RESOURCE_TYPE_APPLICATION_SERVER = 2;
+}
```

Comment:
Like mentioned above, I'd consider removing this and just using existing KindXXX string constants. One less thing to change when adding support for new resource.